ONSdigital/response-operations-ui
# Add js task runner

**Requires orchestrated merge so don't merge for now**
# Motivation and Context
Many of the ends we intend to reach with frontend improvements require the running of tasks, this introduces the task running frameworks.

# What has changed
* Added `node.js`, `npm` setup, and a `gulp` task framework
* Added tasks for every currently intended task to be implemented
* Implemented tasks:
    * CSS Linting
    * SCSS -> CSS compilation
    * JS linting using ESLint
* Other tasks raise a specfic error to point out that they are not yet implemented.

# How to test?
* Run the solution, ensure it works
* Run `gulp scsscompile` to confirm it runs (having followed gulp installation in README)
* Run `docker build .` from the repo to ensure it builds.

# Links
https://trello.com/c/1RVgxFl4
